src/passes/arithmetic/Arithmetic.cpp

```diff
@@ -264,6 +271,14 @@ PreservedAnalyses Arithmetic::run(Module &M,
       Changed |= runOnBasicBlock(BB);
     }
   }
+
+  if (ReportDiff) {
+    llvm::raw_string_ostream(ObfuscatedIR) << M;
+    if (OriginalIR != ObfuscatedIR)
+      config.getUserConfig()->report_diff(name().str(), OriginalIR,
```

Comment:
Think it could be useful to have this wrapped in, e.g., `handleReportDiff` and have it as a utility so as to be used in other passes as well, but for now it's fine as it is.

Reply:
Yes, we need a utility to scale this up to other passes. I added a proposal in [e9735e9](https://github.com/open-obfuscator/o-mvll/pull/49/commits/e9735e9f87d1ad4a84d4ee4179e10d49bdd1898c). It integrates the tracking of explicit `changed` states from individual transformations, which is another crosscutting concern in all passes. What do you think?